mono/mini/debugger-engine.c

```diff
@@ -1610,19 +1610,20 @@ mono_debugger_free_objref (gpointer value)
 MonoClass *
 get_class_to_get_builder_field(DbgEngineStackFrame *frame)
 {
+	StackFrame *the_frame = (StackFrame *)frame;
 	ERROR_DECL (error);
 	gpointer this_addr = get_this_addr (frame);
 	MonoClass *original_class = frame->method->klass;
 	MonoClass *ret;
-	if (!m_class_is_valuetype (original_class) && mono_class_is_open_constructed_type (m_class_get_byval_arg (original_class))) {
```

Comment:
Actually, do you know that `original_class` is a valuetype?   If it's a generic sharing gparam like `T_REF` I think `m_class_is_valuetype (original_class)` will be `FALSE`.   Or maybe it's some class instantiated with `T_REF`? I'm not sure.

I bet you can keep this condition.